recipes/bshoshany-thread-pool/all/conanfile.py

```diff
@@ -38,7 +38,7 @@ def _source_subfolder(self):
         return "source_subfolder"
 
     def package(self):
-        copy(self, "*.hpp", self.source_folder, os.path.join(self.package_folder, "include"))
+        copy(self, "include/*.hpp", self.source_folder, self.package_folder)
```

Comment:
```suggestion
        copy(self, "*.hpp", os.path.join(self.source_folder, "include"), self.package_folder)
```

The second parameter is `pattern` only, not path. That's why you are having the error about empty packages.